### DISCUSSION TITLE
How can I access next-auth session data in getInitialProps?

### DISCUSSION THREAD

**Discussion by LGmatrix13** [vercel/next.js#25160] [Help] [answered]
Hello everyone,

I want to access my next-auth session data in getInitialProps

**My Code:**
```
Account.getInitialProps = async () => {
    const response = await fetch(`http://localhost:3000/api/posts`)
    const items = await response.json()
    return {
      items: items,
    }
  }
```
  
I basically need to change `fetch(`http://localhost:3000/api/posts`)` to `fetch(`http://localhost:3000/api/posts`+session.name)`

Any suggestions? 
Thank you!

**Comment by wwwdepot** [comment, marked as answer]
Use the getSession method for server-side

https://next-auth.js.org/tutorials/securing-pages-and-api-routes#server-side

```js
import { getSession } from 'next-auth/client'

Account.getInitialProps = async (ctx) => {
    const session = await getSession(ctx)

    ...
}
```

**Reply by LGmatrix13** [reply]
@wwwdepot thank you!
